src/app/modules/shared-components/components/switch/switch.component.html

```diff
@@ -1,11 +1,23 @@
-<p-inputSwitch
-  #switch
-  [(ngModel)]="checked"
+<p-inputSwitch *ngIf="parentForm && formControl; else notReactive"
+  [formControl]="formControl"
   [ngClass]="{
     dark: mode === 'dark',
     circular: mode === 'circular',
     white: mode === 'white',
     'dark-circular': mode === 'dark-circular'
   }"
-  (onChange)="handleChange(switch.checked)"
 ></p-inputSwitch>
+
+<ng-template #notReactive>
+  <p-inputSwitch
+    #switch
+    [(ngModel)]="checked"
+    [ngClass]="{
+      dark: mode === 'dark',
+      circular: mode === 'circular',
+      white: mode === 'white',
+      'dark-circular': mode === 'dark-circular'
+    }"
+    (onChange)="handleChange(switch.checked)"
+  ></p-inputSwitch>
+</ng-template>
```

Comment:
Why not... but it needs to be clear. You should add documentation to the class explaining that either parentForm and a valid "name" in the form are given, either output will be emitted, but not both.

Reply:
fixed